### PR TITLE
[ticket/16629] Fix ACP get_database_size() for MySql 8

### PR DESCRIPTION
**Pull request by 3D-I**
PHPBB3-16629

Checklist:

- [x] Correct branch: master for new features; 3.3.x & 3.2.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/dev/master/development/coding_guidelines.html), [3.3.x](https://area51.phpbb.com/docs/dev/3.3.x/development/coding_guidelines.html) and [3.2.x](https://area51.phpbb.com/docs/dev/3.2.x/development/coding_guidelines.html)
- [x] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.3.x/development/git.html)

https://tracker.phpbb.com/browse/PHPBB3-16629

### CODE REVIEW

File: phpBB/includes/functions_admin.php [hunk 1]
```diff
@@ -2853,35 +2851,25 @@ function get_database_size()
...
+				$sql = 'SHOW TABLE STATUS
+					FROM ' . $db_name;
+				$result = $db->sql_query($sql, 7200);
 
-					$database_size = 0;
-					while ($row = $db->sql_fetchrow($result))
+				while ($row = $db->sql_fetchrow($result))
+				{
+					if (isset($row['Engine'])
+						&& ($row['Engine'] == ('MyISAM' || 'InnoDB' || 'Aria'))
```
Comment (MioVisman) [inline phpBB/includes/functions_admin.php:2863]: Very interesting comparison for php

Comment (3D-I) [reply phpBB/includes/functions_admin.php]: Got confused by `arbitrary expressions in empty`, my bad.

**Comment by marc1706** [comment]
I don't have the described issue in my MySQL8 instance. Can you elaborate a bit more on how this issue can be reproduced and how you're solving this here?

**Comment by 3D-I** [comment]
The old 
`if (preg_match('#(3\.23|[45]\.|10\.[0-9]\.[0-9]{1,2}-+Maria)#', $version))` 
simply does not recognise the `$row['mysql_version']` which in my case is `string '8.0.20'` AKA **8.0.20**. 

You can test it here https://www.regexpal.com/?fam=118322
Moreover it is very old and useless at this stage _(who needs to check for versions 3 and 4 nowadays?)_, same applies for the second preg_match.

It is also useless the IF_ELSE statement below `if ($table_prefix != '')` since right or wrong it returns the same sum of vars:
 `$database_size += $row['Data_length'] + $row['Index_length'];`

BEFORE
immagine

AFTER
immagine

**Review by marc1706 — CHANGES_REQUESTED** [review]

**Review by marc1706 — APPROVED** [review]

**Pull request merged**